### PR TITLE
build: add nix dev + build environment + ci builds

### PR DESCRIPTION
**Pull request by mrcjkb**
The obligatory flakeification :smile:

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,83 @@
+{
+  description = "devShell for Rust projects";
```
Comment (teto) [inline flake.nix:2]: thats a bit generic

Comment (mrcjkb) [reply flake.nix:2]: Oh yeah, that's from my template...

**Comment by mrcjkb** [comment]
@teto for some reason, the extra `buildInputs` I add to the devShell (`rust-analyzer`, `rustfmt`, ...) aren't available on the `PATH`.
Any idea what could be the issue?

**Comment by teto** [comment]
I am never too sure about the rules but because it's not a mkShell but the program buildInputs, in which case you need propagatedBuildInputs to propagate to PATH .
Maybe if you append `.inputDerivation` at the end ?

**Comment by mrcjkb** [comment]
> I am never too sure about the rules but because it's not a mkShell but the program buildInputs, in which case you need propagatedBuildInputs to propagate to PATH . Maybe if you append `.inputDerivation` at the end ?

:thinking: none of it worked. Maybe it's something that `rustPlatform.buildRustPackage` does...
I ended up using `mkShell` and appending `rocks.buildInputs ++ rocks.nativeBuildInputs` to the shell's `buildInputs`.
A bit hacky, but it works :shrug:

**Review by teto — APPROVED** [review]
I've had similar issues in the past so I get it. Approved

**Comment by mrcjkb** [comment]
Oops... forgot to set it to squash...

**Pull request merged**